File: app/src/main/java/com/coder/zzq/smartshowdemo/MainActivity.java
package com.coder.zzq.smartshowdemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import com.coder.zzq.smartshow.dialog.ClickListAdapter;

import java.util.Arrays;

public class MainActivity extends AppCompatActivity {
    private ListView mListView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mListView = findViewById(R.id.list_view);
        ClickListAdapter adapter = new ClickListAdapter();
        adapter.setItemCenter(true);
        adapter.setItems(Arrays.asList(
                new String[]{
                        "toast",
                        "type toast",
                        "snackbar",
                        "topbar",
                        "dialog"
                }
        ));
        mListView.setAdapter(adapter);
        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                switch (position) {
                    case 0:
                        onToastClick(view);
                        break;
                    case 1:
                        onTypeToastClick(view);
                        break;
                    case 2:
                        onSnackbarClick(view);
                        break;
                    case 3:
                        onTopBarClick(view);
                        break;
                    case 4:
                        onDialogClick(view);
                        break;
                }
            }
        });
    }

    public void onToastClick(View view) {
        startActivity(new Intent(this, TestToastActivity.class));
    }

    public void onSnackbarClick(View view) {
        startActivity(new Intent(this, TestSnackbarActivity.class));
    }

    public void onTopBarClick(View view) {
        startActivity(new Intent(this, TestTopbarActivity.class));
    }

    public void onTypeToastClick(View view) {
        startActivity(new Intent(this, TestTypeToastActivity.class));
    }

    public void onDialogClick(View view) {
        startActivity(new Intent(this, TestDialogActivity.class));
    }
}

File: app/src/main/java/com/coder/zzq/smartshowdemo/MainActivity.kt
package com.coder.zzq.smartshowdemo

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import androidx.appcompat.app.AppCompatActivity
import com.coder.vincent.smart_dialog.click_list.ClickListAdapter
import com.coder.vincent.smart_snackbar.SmartSnackBar
import com.coder.zzq.smartshowdemo.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(ActivityMainBinding.inflate(layoutInflater).apply {
            val location = IntArray(2)
            findViewById<View>(android.R.id.content).getLocationOnScreen(location)
            val adapter = ClickListAdapter()
            adapter.setItemCenter(itemCenter = true, notify = false)
            adapter.setItems(
                listOf(
                    "classic toast",
                    "emotion toast",
                    "snackbar",
                    "dialog"
                ), false
            )
            listView.adapter = adapter
            listView.onItemClickListener =
                OnItemClickListener { _, _, position, _ ->
                    when (position) {
                        0 -> onToastClick()
                        1 -> onTypeToastClick()
                        2 -> onSnackBarClick()
                        3 -> onDialogClick()
                    }
                }
        }.root)
    }

    fun onToastClick() {
        startActivity(Intent(this, ClassicToastDemoActivity::class.java))
    }


    private fun onSnackBarClick() {
        SmartSnackBar.top(this)
            .config()
            .actionLabelColor(Color.CYAN)
            .actionLabelColorResource(R.color.colorAccent)
            .actionLabelSizeSp(15f)
            .actionLabelBold(true)
            .apply()
            .show("setup action text style")

    }

    fun onTypeToastClick() {
        startActivity(Intent(this, EmotionToastDemoActivity::class.java))
    }

    private fun onDialogClick() {
        startActivity(Intent(this, TestDialogActivity::class.java))
    }
}